### ISSUE TITLE
Color variables needs to be reviewed

### ISSUE THREAD

**Issue by mparvazi** [PrestaShop/hummingbird#186]
**Current color overrides:**

`src/scss/partials/variables/bootstrap/_variable-overrides.scss`
| Sass | Color | Contrast Ratio* | WCAG AA | BS5 !default |
| :---         |     :---:      |    :---:      | :---         |    :---:      |
| $primary   | `#5434d0`   | 7.55:1   | Pass   | `#0d6efd`   |
| $secondary   | `#f39d72`   | 9.87:1   | Pass   | `#6c757d`   |
| $Info**   | `#f39d72`   | 9.87:1   | Pass   | `#0dcaf0`   |
| $success   | `#4cbb6c`   | 2.43:1   | Fail   | `#198754`   |
| $warning   | `#ff9a52`   | 9.99:1   | Pass   | `#ffc107`   |
| $danger   | `#ff4c4c`   | 3.28:1   | Fail   | `#dc3545`   |

`src/scss/partials/variables/prestashop/_colors.scss`
| Sass | Color | Contrast Ratio* | WCAG AA | BS5 !default |
| :---         |     :---:      |    :---:      | :---         |    :---:      |
| $default-icon-color   | `#6c868e`   | 3.86:1   | Fail   | -   |
| $footer-background-color   | `#011638`   | 17.9:1   | Pass   | -   |

\* _For [contrast checking](https://webaim.org/resources/contrastchecker/) I used colors as background and white or black as foreground color._
** _The **$info** color basically should be in blue color tone._

**These files have been used the hard coded colors:**
```
src/scss/error.scss
src/scss/core/components/_breadcrumb.scss
src/scss/core/modules/nouislider/_vars.scss
src/scss/custom/components/_order-confirmation.scss
src/scss/custom/components/customer/_customer-menu.scss
src/scss/custom/layout/_footer.scss
```

## BS5

**gray-color-variables**
| Sass | !default |
| :---         |     :---:      |
|$white:   |`#ffffff`|
|$gray-100 |`#f8f9fa`|
|$gray-200 |`#e9ecef`|
|$gray-300 |`#dee2e6`|
|$gray-400 |`#ced4da`|
|$gray-500 |`#adb5bd`|
|$gray-600 |`#6c757d`|
|$gray-700 |`#495057`|
|$gray-800 |`#343a40`|
|$gray-900 |`#212529`|
|$black:    |`#000000`|

**color-variables**
| Sass | !default |
| :---         |     :---:      |
|$blue    |`#0d6efd`|
|$indigo  |`#6610f2`|
|$purple  |`#6f42c1`|
|$pink    |`#d63384`|
|$red     |`#dc3545`|
|$orange  |`#fd7e14`|
|$yellow  |`#ffc107`|
|$green   |`#198754`|
|$teal    |`#20c997`|
|$cyan    |`#0dcaf0`|

**Comment by adamus007p** [comment]
@mparvazi is there any idea to create a dark, white theme which this two options will be by default?

**Comment by mparvazi** [comment]
> @mparvazi is there any idea to create a dark, white theme which this two options will be by default?

BS5 is ready to use for light/dark theming and there is many options for it.

As a simple example:

```
:root {
  --content-text-color: $gray-900;
}

body.dark {
  --content-text-color: $gray-100;
}
```

```
p {
  color: var(--content-text-color);
}
```

By default `$gray-900` will be used for text-color, but when you add `.dark` class to `<body>` then `$gray-100` will be used.

You just need to add Sass color variables into maps, and then make them ready for light/dark theming.

```
$colors: (
  $primary: $dark;
);

$dark-colors: (
  $primary: $light;
);

:root {
  @each $color, $value in $colors {
    --#{$variable-prefix}#{$color}: #{$value};
  }
}

body.dark {
  @each $color, $value in $dark-colors {
    --#{$variable-prefix}#{$color}: #{$value};
  }
}
```

**Comment by mparvazi** [comment]
@NeOMakinG @prestascott 

IMO we need to just modify **gray-color-variables** and **color-variables** at first.
Then we can use this vars for components. 

For example the current primary color is near to `$indigo` at BS5 default.

$indigo: `#5434d0`;
$primary: $indigo;

I think something like this is not correct:
$primary: `#5434d0`;

**Comment by NeOMakinG** [comment]
Thanks for the investigations, to be decided with @prestascott

**Comment by NeOMakinG** [comment]
@mparvazi I heard that colors will change in the future, to be more generic and less linked to the PrestaShop colors, let's wait for the @prestascott response :)

**Comment by prestascott** [comment]
Yes, some colors will change in order to have a more neutral theme. For example, the primary button won't be purple anymore but almost-black color.

**Comment by SharakPL** [comment]
Would be best to also provide configuration in admin panel to set main theme color. It could be used eg. in mail templates to better match current theme main color. Now they use default blue from Classic theme and if you want to change it you have to edit every single mail template.

**Comment by NeOMakinG** [comment]
> Would be best to also provide configuration in admin panel to set main theme color. It could be used eg. in mail templates to better match current theme main color. Now they use default blue from Classic theme and if you want to change it you have to edit every single mail template.

I think it's related to the core system, not the theme, feel free to open an issue on `PrestaShop/PrestaShop` about this if you want this feature, so the product team can think about it!

Or, it could be a module, but I can't decide alone if we want to maintain this

**Comment by NeOMakinG** [comment]
@mparvazi https://www.figma.com/file/LfVl5leeSKcVUhSaYwhbtM/New-Theme?node-id=200%3A9948

Scott pushed new colors to the Figma, we need to work on the theme about this!

**Comment by mparvazi** [comment]
> @mparvazi https://www.figma.com/file/LfVl5leeSKcVUhSaYwhbtM/New-Theme?node-id=200%3A9948
> 
> 
> 
> Scott pushed new colors to the Figma, we need to work on the theme about this!

IMO, black color as primary make the theme too official.

**Comment by SharakPL** [comment]
We should provide configuration to set at least primary and maybe secondary color (obviously not in this repo).
Hummingbird's template should inline something like this after theme.css
```smarty
{if isset($themeColors)}
  <style>
    :root {ldelim}
      --bs-primary: {$themeColors.primary};
      --bs-secondary: {$themeColors.secondary}; {* optional *}
    {rdelim}
  </style>
{/if}
```
It could be small module to handle this or another core functionality in admin panel, but either way the theme would have to start using css variables instead of hardcoded colors.

**Comment by NeOMakinG** [comment]
> We should provide configuration to set at least primary and maybe secondary color (obviously not in this repo). Hummingbird's template should inline something like this after theme.css
> 
> ```
> {if isset($themeColors)}
>   <style>
>     :root {ldelim}
>       --bs-primary: {$themeColors.primary};
>       --bs-secondary: {$themeColors.secondary}; {* optional *}
>     {rdelim}
>   </style>
> {/if}
> ```
> 
> It could be small module to handle this or another core functionality in admin panel, but either way the theme would have to start using css variables instead of hardcoded colors.

Honestly, this is a goal, we should replace every variable into CSS Variables, it's something I really want before the theme gets stable because it permits anyone to modify these colors without any transpiler/bundler

**Comment by NeOMakinG** [comment]
> > @mparvazi https://www.figma.com/file/LfVl5leeSKcVUhSaYwhbtM/New-Theme?node-id=200%3A9948
> > Scott pushed new colors to the Figma, we need to work on the theme about this!
> 
> IMO, black color as primary make the theme too official.

Agree, but it's probably the simplest way to have a neutral theme, not linked to the PrestaShop company colors :/

**Comment by SharakPL** [comment]
>  
> But there are some limitations about this on bootstrap side too

True, but soon all BS components will support this https://github.com/twbs/bootstrap/issues/35319 and so should hummingbird.

**Comment by NeOMakinG** [comment]
> twbs/bootstrap#35319

Agree, can't wait !

I think we can wait for the 5.3 to be released without any problems
